include/openthread/error.h

```diff
@@ -236,6 +236,21 @@ typedef enum OT_MUST_USE_RESULT otError
      */
     OT_ERROR_REJECTED = 37,
 
+    /**
+     * The channel provided is not supported.
+     */
+    OT_ERROR_UNSUPPORTED_CHANNEL = 38,
+
+    /**
+     * The action didn't succeed because Thread is in Disabled state.
+     */
+    OT_ERROR_THREAD_DISABLED = 39,
+
+    /*
+     * The action didn't succeed because some precondition isn't satisfied.
+     */
+    OT_ERROR_FAILED_PRECONDITION = 40,
+
```

Comment:
I would suggest reusing any existing error codes if possible (it would be safer). In my view, we already have too many error codes (which are specifically used in some cases). I would prefer to have more common generic errors that are shared and may be interpreted based on their use.

For example, how about `INVALID_STATE` (or `OT_ERROR_DETACHED`) instead of `DISABLED`?

Otherwise, it may be fine to add these. But then I would suggest we add them in `error.hpp` and ensure `ErrorToString()`/`otThreadErrorToString()` will return the correct string for them.

Reply:
That makes sense! Actually I'm not a fan of adding more error codes in OT for specific cases neither. The errors added here are not for OT APIs, but for [ThreadHost APIs](https://github.com/openthread/ot-br-posix/blob/main/src/ncp/thread_host.hpp#L118) in ot-br-posix. The tricky thing is that the existing client code (like dbus, android binder server) has a broad usage on `otError`. So `otError` has to be used.

After thinking more this, I plan to add a new error code in ThreadHost which is a super set of otErrors. It could be a signed int. We can add ThreadHost specific errors to this new type. And existing client code could also work (comparing this new error code with OT_ERROR_XXX).